CIP-0155/README.md

```diff
@@ -0,0 +1,133 @@
+---
+CIP: 0155
+Title: SRV registry
+Status: Proposed
+Category: Network
+Authors:
+  - Marcin Szamotulski <marcin.szamotulski@iohk.io>
+Implementors: N/A
+Discussions: 
+  - [PR](https://github.com/cardano-foundation/CIPs/pull/1033)
+Created: 2025-04-22
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This CIP defines the procedure for assigning Service Record (SRV) prefixes for Cardano applications (like `cardano-node`, `mithril`, etc.).
+
+This means of sharing authoritative information between SPOs who deploy services and _decentralised protocol_ (DP) developers who write them.
+
+## Motivation: why is this CIP necessary?
+
+The **Cardano Ledger** allows the use of SRV records in the **SPO** registration certificate (see [register-stake-pool]).
+
+In this CIP, we propose to make them usable by decentralised applications running on `Cardano`.
+
+Having access to the ledger (either directly or through tools like `cardano-cli`) will allow decentralised protocols developers to construct networks based on registered relays. 
+
+All involved parties (SPOs, DP developers, etc) need to agree on SRV prefixes used by various DPs and thus a public registry governed by a public CIP process is required to foster DPs that co-exist with the **Cardano** network, like **Mithril** (see [CIP#137]).
+
+We consider DPs as applications, which construct their own network and relay on ledger peers published on the block-chain. This include node implementations, or 2nd-layer solutions like Mithril or Hydra.
+
+### Problem
+
+Bootstraping networks requires sharing information about available services.
+
+Historically, this was done by sharing IP or DNS names and PORT numbers, but this approach has its limitations.
+
+SRV records were designed to make deployment independent of hard-coded service end-points (see [RFC#2782]).
+
+Thus they can be instrumental in `Cardano`, since relays are registered on the blockchain through the registration certificate.
+
+By using SRV records in the registration certificate (which is supported by the `cardano-ledger`, but not by `cardano-node`), we wish to solve this problem not just for `Cardano` node implementation, but also for any DP that requires constructing its own network.
```

Comment:
What does it mean that this is supported by the `cardano-ledger` but not the `cardano-node`? As in, there's a field in the SPO registration certificate for it, but the node doesn't actually use that to make SRV requests for peer discovery?

Reply:
> As in, there's a field in the SPO registration certificate for it, but the node doesn't actually use that to make SRV requests for peer discovery?

Yes, you got it right :).